### PR TITLE
feat(pegboard): integrate setup scripts

### PR DESCRIPTION
**Pull request by MasterPtato**

### CODE REVIEW

**Comment by MasterPtato** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1147?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1147** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1147?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1146** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1146?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1145** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1144** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1144?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1143** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1143?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1142** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1142?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1141** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1141?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MasterPtato and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/pegboard/container-runner/src/main.rs [hunk 1]
```diff
@@ -33,6 +33,12 @@ fn main() -> anyhow::Result<()> {
 		.context("`container_dir` arg required")?;
 	let pegboard_container_dir = Path::new(&pegboard_container_dir);
 
+	// Write PID to file
+	fs::write(
+		pegboard_container_dir.join("pid"),
```
Comment (NathanFlurry) [inline lib/pegboard/container-runner/src/main.rs:38]: nit: use consts for all files that are written. makes it easy to know what's in the folder

File: lib/pegboard/manager/src/ctx.rs [hunk 2]
```diff
@@ -207,6 +226,8 @@ impl Ctx {
 		// Check file doesn't exist
 		if fs::metadata(&path).await.is_err() {
 			utils::download_file(container_runner_binary_url, &path).await?;
+
+			Command::new("chmod").arg("+x").arg(&path).output().await?;
```
Comment (NathanFlurry) [inline lib/pegboard/manager/src/ctx.rs:230]: nit: use fs api for this

File: lib/pegboard/container-runner/src/main.rs (hunk 1 above)
Comment (MasterPtato) [reply lib/pegboard/container-runner/src/main.rs:38]: could you elaborate?

**Pull request closed without merging**